test/cqlpy/test_tablets.py

```diff
@@ -181,7 +181,40 @@ def test_describe_initial_tablets(cql, skip_without_tablets):
             "AND TABLETS = { 'initial' : 1 }"
     with new_test_keyspace(cql, ksdef) as keyspace:
         desc = cql.execute(f"DESCRIBE KEYSPACE {keyspace}")
-        assert "and tablets = {'initial': 1}" in desc.one().create_statement.lower()
+        assert "and tablets = {'enabled': true, 'initial': 1}" in desc.one().create_statement.lower()
+
+
+# Test to ensure metadata is printed when tablets are enabled during schema description, regardless of the use of WITH INTERNALS.
+#
+# Reproduces https://github.com/scylladb/scylladb/issues/22866
+@pytest.mark.parametrize("tablets", [None, True, False])
+@pytest.mark.parametrize("initial", [None, 0, 7])
+def test_describe_tablets(cql, tablets, initial):
+    if not tablets and initial is not None:
+        pytest.skip('The combination of disabled tablets and initial having a value makes no sense')
+    tablets_str = (f"'enabled': {tablets}" if tablets is not None else "").lower()
```

Comment:
I'm understand something new now... It seems from this that if "tablets" is not specified (tablets=None in your test), then Scylla chooses either tablets or vnodes depending on its default, but DESC TABLE does **not** print tablets=true or tablets=false so you don't know what it chose. Is this deliberate?
If that's the case then I agree it's good testing it, at least to prevent an accidental regression from changing this decision - regardless of whether it's a good or bad decision.

Reply:
>but DESC TABLE does not print


Not sure I'm following, if one chooses not to specify `tablets=true` when creating the keyspace then default is used, when you `DESCRIBE` the keyspace and default is set to use tablets it will be reflected in the describe result, take a look at the cover letter, it is the first case. If the default is to use `vnodes` then it will print nothing in the describe output, does it crarify things?

Reply:
Ah, now I see
>if it doesn't test anything
bug in the test, fixing
>even if "with tablets" wasn't given in the input
the current implementation prints "tablets enabled=true" only in the case the ferature is enabled, once it is not the `describe` doesnt mention tablets at all, like "tablets enabled=false"

Reply:
moreover, I think this test does not relevant to the case when tablets are disabled on cluster level, so I will add `skip_without_tablets` fixture to it